internal/database/types_hcpcluster.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+)
+
+type CustomerDesiredHCPClusterState struct {
+	// HCPOpenShiftCluster contains the desired state from a customer.  It is filtered to only those fields that customers
+	// are able to set.
+	// We will whitelist specific fields as we go through to prevent conflicts.
```

Comment:
I think I'll need you to clarify what you have in mind with the filter/whitelist comments.  It sounds like it might overlap with some validation checks we already do.